From: Bastrykov Evgeniy <evgeny.bastrykov@flant.com>
Subject: Disable registration using the attribute constructor

Since we use runtime select of implementation based on CPU features. We need to control
which implementation should be registered.
---
diff --git a/src/jemalloc.c b/src/jemalloc.c
index 8d341ba..d456d86 100644
--- a/src/jemalloc.c
+++ b/src/jemalloc.c
@@ -239,7 +239,6 @@ static malloc_mutex_t init_lock = SRWLOCK_INIT;
 static malloc_mutex_t init_lock;
 static bool           init_lock_initialized = false;
 
-JEMALLOC_ATTR(constructor)
 static void WINAPI
 _init_init_lock(void) {
 	/*
@@ -3299,8 +3298,7 @@ label_done:
  * via a library constructor that runs before jemalloc's runs.
  */
 #ifndef JEMALLOC_JET
-JEMALLOC_ATTR(constructor)
-static void
+void
 jemalloc_constructor(void) {
 	malloc_init();
 }
diff --git a/src/zone.c b/src/zone.c
index e09de4b..f09ad82 100644
--- a/src/zone.c
+++ b/src/zone.c
@@ -435,7 +435,6 @@ zone_promote(void) {
 	} while (zone != &jemalloc_zone);
 }
 
-JEMALLOC_ATTR(constructor)
 void
 zone_register(void) {
 	/*
